lib/iris/fileformats/grib/__init__.py

```diff
@@ -692,6 +688,10 @@ def _message_values(grib_message, shape):
     data = gribapi.grib_get_double_array(grib_message, 'values')
     data = data.reshape(shape)
 
+    # Handle missing values in a sensible way.
+    mask = np.isnan(data)
+    if mask.any():
+        data = ma.array(data, mask=mask, fill_value=np.nan)
```

Comment:
@lbdreyer Just confirming my understanding ... so this is change is making up for the removal of `convert_nans_array` to convert an array containing nans to a masked array, got it ... but you're explicitly setting the `fill_value` to be nan, which we didn't ever do before. Why so?

Also, this function `_message_values` is called from the `GribWrapper.__init__`, but also the `GribDataProxy.__getitem__` ... which is interesting as the GribDataProxy didn't ever then apply `convert_nans_array` ...